sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SimpleSyncPoller.java

```diff
@@ -0,0 +1,284 @@
+        PollingContext<T> context1 = this.pollingContext.copy();
+        if (context1.getActivationResponse() == context1.getLatestResponse()) {
+            this.cancelOperation.apply(context1, context1.getActivationResponse());
+        } else {
+            try {
+                this.cancelOperation.apply(null, this.activationResponse);
+            } catch (PollContextRequiredException crp) {
+                PollingContext<T> context2 = this.pollingContext.copy();
+                pollingLoopSync(context2);
+                this.cancelOperation
+                    .apply(context2, this.activationResponse);
+            }
+        }
+    }
+
+    @Override
+    public SyncPoller<T, U> setPollInterval(Duration pollInterval) {
+        Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
+        if (pollInterval.isNegative() || pollInterval.isZero()) {
+            throw LOGGER.logExceptionAsWarning(new IllegalArgumentException(
+                "Negative or zero value for 'pollInterval' is not allowed."));
+        }
+        this.pollInterval = pollInterval;
+        return this;
+    }
+
+    private static <T, U> PollResponse<T> toPollResponse(SyncPollResponse<T, U> syncPollResponse) {
+        return new PollResponse<>(syncPollResponse.getStatus(),
+            syncPollResponse.getValue(),
+            syncPollResponse.getRetryAfter());
+    }
+
+    private boolean matchStatus(AsyncPollResponse<T, U> currentPollResponse,
+                                LongRunningOperationStatus statusToWaitFor) {
+        if (currentPollResponse == null || statusToWaitFor == null) {
+            return false;
+        }
+        if (statusToWaitFor == currentPollResponse.getStatus()) {
+            return true;
+        }
+        return false;
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext) {
+        return this.pollingLoopSync(pollingContext, Optional.empty(), Optional.empty());
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, LongRunningOperationStatus statusToWaitFor) {
+        return this.pollingLoopSync(pollingContext, Optional.empty(), Optional.of(statusToWaitFor));
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, Duration timeout,
+                                                   LongRunningOperationStatus statusToWaitFor) {
+        return this.pollingLoopSync(pollingContext, Optional.of(timeout), Optional.of(statusToWaitFor));
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, Duration timeout) {
+        return this.pollingLoopSync(pollingContext, Optional.of(timeout), Optional.empty());
+    }
+
+    private SyncPollResponse<T, U> pollingLoopSync(PollingContext<T> pollingContext, Optional<Duration> timeout,
+                                                   Optional<LongRunningOperationStatus> statusToWaitFor) {
```

Comment:
Why do we use `Optional` for nullable types that should could just be checked for `!= null`?